src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -521,6 +535,15 @@ private Task ProcessUnknownFrameAsync()
                 await OnEndStreamReceived();
             }
 
+            if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+            {
+                // All HTTP/3 requests MUST include exactly one valid value for the :method, :scheme, and :path pseudo-header
```

Comment:
Is this not a requirement for HTTP/2?

Reply:
HTTP/2 and HTTP/3 have the same requirements here.